examples/mnist.py

```diff
@@ -1,10 +1,20 @@
 from __future__ import print_function
 
-import cPickle as pickle
 import gzip
 import itertools
-import urllib
+import pickle
 import os
+import sys
+
+PY2 = sys.version_info[0] == 2
+
+if PY2:
+    from urllib import urlretrieve
+    pickle_load = lambda f, encoding: pickle.load(f)
+else:
+    from urllib.request import urlretrieve
+    pickle_load = lambda f, encoding: pickle.load(f, encoding=encoding)
+
```

Comment:
You're right that `pickle` is faster after the cache has been warmed.  But you can understand my suspicion due to the results from these first two (un-warmed) lines:

```
>>> from timeit import timeit
>>> timeit("pickle.load(gzip.open('mnist.pkl.gz', 'rb'))", setup="import pickle; import gzip", number=10)
5.902479887008667
>>> timeit("pickle.load(gzip.open('mnist.pkl.gz', 'rb'))", setup="import cPickle as pickle; import gzip", number=10)
5.870376825332642
>>> timeit("pickle.load(gzip.open('mnist.pkl.gz', 'rb'))", setup="import pickle; import gzip", number=10)
5.449158191680908
>>> timeit("pickle.load(gzip.open('mnist.pkl.gz', 'rb'))", setup="import cPickle as pickle; import gzip", number=10)
5.793623924255371
```

However, I guess that warming shouldn't even be an issue when I and [others](http://justinfx.com/2012/07/25/python-2-7-3-serializer-speed-comparisons/) have observed orders of magnitude speed differences between the two.  I wonder what's going on here.